src/snap/invokeNotification.ts

```diff
@@ -0,0 +1,47 @@
+import { DappeteerPage, Serializable } from "../page";
+import { clickOnElement, openProfileDropdown } from "../helpers";
+import { invokeSnap } from "./invokeSnap";
+
+async function waitForNotification(page: DappeteerPage): Promise<void> {
```

Comment:
this won't work unless the user navigates to the notifications screen. You can't expect the user to call invokeNotification before this method.

Use code of invokeNotification here and remove invoke notification method,

Reply:
It's an internal method for invokeNotification. I think the API is pretty straightforward, it exactly solves the problem of the user having some additional knowledge about how notifications will be rendered and it's not clear that after invoking the notification before getting them all you need to call an additional method to wait until it's renders in the DOM. I think it's pretty simple the way it works right now and all the complex logic of awaiting and opening the pages is hidden from the user:

```
invokeNotification()
getAllNotifications()
```

We can do waitForNotification with invokeMethod, but the user has to implement more details
```
openNotificationPage() // as we need to open the page where we are waiting for the methods
invokeMethod() // notification call
waitForNotification() //  waits until rendered
getAllNotifications 
```

I would prefer to have one method which will solve most user needs to get the notifications texts and not force them to implement something additional. If I need custom logic, I can still use invokeMethod to call a notification

Reply:
I don't think it's clear at all.Why are you not implementing following:
- waitForNotification
   - open new tab
   - navigate to notifications screen
   - wait for notification
   - before resolving with new notification close tab
- getAllNotifications
   - navigate metamask screen to navigation
   - obtain all notification   
   - redirect to metamask home screen

Which provides common API and flows:
```typescript
const notificationPromise = waitForNotification()
await invokeSnap()
const notification = await notificationPromise
const notiications = await getAllNotifications()
```
Everything is easy to use, self-contained and intuitive.

Reply:
@mpetrunic I tried to implement the way you suggested, the problem here is when we call waitForNotification synchronously and invoke a snap the page with the notification is not opened yet (the browser opens a new tab takes a lot of time), so when we get to the point of waiting for notification Promise, the notification already executed, but we did not get to the notification page where we listen for the DOM changes. It could be solved by opening the page first and then calling to wait for notification, we should have a proper order.

```js
await openNotificationPage()
const notificationPromise = waitForNotification()
await invokeSnap()
const notification = await notificationPromise
const notiications = await getAllNotifications()
```
openNotificationPage
- open new tab
- navigate to notifications screen

waitForNotification
- wait for notification
- before resolving with new notification close tab

getAllNotifications
- navigate metamask screen to navigation
- obtain all notification
- redirect to metamask home screen


Also, How do you imagine the case with two notifications? 
```js
await openNotificationPage()

const notificationPromise = waitForNotification()
await invokeSnap()
const notification = await notificationPromise

const notificationPromise2 = waitForNotification()
await invokeSnap()
const notification = await notificationPromise2

const notiications = await getAllNotifications()
```
Are you fine with such an implementation?

Reply:
I'm more for creating a static class that extends emitter and has all that methods for getting all notifications waiting, etc...

Reply:
hmm but you need to invoke some async method that would prepare new tab with notifications screen and event emitter doesn't provide that kind of initializer